sfn-query-large-datasets-cdk/README.md

```diff
@@ -0,0 +1,110 @@
+
+* An Amazon Athena query
+* An AWS Glue crawler
+* AWS Lambda Functions
+* An Amazon Simple Notification Service topic
+* Related AWS Identity and Access Management (IAM) roles
+
+Learn more about this workflow at Step Functions workflows collection: https://docs.aws.amazon.com/step-functions/latest/dg/sample-query-large-datasets.html
+
+Important: this application uses various AWS services and there are costs associated with these services after the Free Tier usage - please see the [AWS Pricing page](https://aws.amazon.com/pricing/) for details. You are responsible for any AWS costs incurred. No warranty is implied in this example.
+
+## Requirements
+
+* [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls and manage AWS resources.
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) (AWS SAM) installed
+
+## Deployment Instructions
+
+1. If this is your first time using AWS CDK, bootstrap your [environment](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html#getting_started_bootstrap).
+    ```
+    cdk bootstrap aws://{your-aws-account-number}/{your-aws-region}
+    ```
+
+2. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+    ``` 
+    git clone https://github.com/aws-samples/step-functions-workflows-collection
+    ```
+3. Change directory to the pattern directory:
+    ```
+    cd sfn-query-large-datasets-cdk
+    ```
+4. Create a Python virtual environment and install the requirements:
+    ```
+    python3 -m venv .venv
+    source .venv/bin/activate
+    python3 -m pip install -r requirements.txt
+    ```
+5. From the command line, use CDK to deploy the AWS resources for the workflow as specified in the ```app.py``` file:
+    ```
+    cdk deploy
+    ```
+6. During the prompts:
+    * ```Do you wish to deploy these changes (y/n)? Y ```
+
+## How it works
+
+The workflow starts by initializing a state machine, and the first task in the sequence is to ingest a large data set in Amazon S3 and partition it through aws Glue Crawlers.
+
+The Athena query is then initiated and runs until it is completed.
+
+Once the query is completed, the results are obtained and published to the SNS (Simple Notification Service) topic. SNS is a web service that allows users to send notifications from the cloud to various subscribers or endpoints, such as email, SMS, or mobile push notifications.
```

Comment:
Not sure I like "SNS is a web service". I'd rather say "SNS is a fully managed service"